doc/sphinx/language/extensions/match.rst

```diff
@@ -244,6 +244,33 @@ Printing of the elimination predicate
    the result type is not printed when Coq knows that it can re-
    synthesize it.
 
+Printing of hidden subterms
++++++++++++++++++++++++++++
+
+.. flag:: Printing Match All Subterms
+
+   In order to be able to cheaply reconstruct the types of the
+   variables bound by `in` and `as`, `match` terms contain the
+   polymorphic universe instance and the parameters of the inductive
+   which is being matched. When this flag is on (it is off by
+   default), this information is displayed as a volatile cast around
+   the match discriminee.
+
+   When the match relies on :flag:`Definitional UIP` the indices are also subterms of the `match` term and are displayed when this flag is on.
+   Otherwise they are not subterms and are displaye as holes `_` when this flag is on.
+
+   .. coqtop:: all
```

Comment:
```suggestion
   .. example:
   .. coqtop:: in
```

Then indent all of the example lines by another 3 spaces.  Use "in" for the first couple lines to make the example shorter.